.github/workflows/publish.yaml

```diff
@@ -0,0 +1,17 @@
+# A CI configuration to auto-publish pub packages.
+
+name: Publish
+
+on:
+  pull_request:
+    branches: [ master ]
+  push:
+    tags: [ 'v[0-9]+.[0-9]+.[0-9]+' ]
```

Comment:
Historically we have not used the `v` prefix on tags in this repo. I'm happy to start using that pattern here, but I can't recall if we've made specific decisions to make the patterns standard.